app/routers/notification.py

```diff
@@ -0,0 +1,167 @@
+from operator import attrgetter
+from typing import List, Union
+
+from fastapi import APIRouter, Depends, Request
+from fastapi.responses import RedirectResponse
+from sqlalchemy.exc import SQLAlchemyError
+from sqlalchemy.orm import Session
+from starlette.status import HTTP_302_FOUND
+
+from app.database.models import Invitation, Message
+from app.dependencies import templates, get_db
+from app.internal.utils import mark_as_read, create_model
+from app.routers.share import accept, decline
+
+NOTIFICATION_TYPE = Union[Invitation, Message]
+
+
+router = APIRouter(
+    prefix="/notification",
+    tags=["notification"],
+    dependencies=[Depends(get_db)]
+)
+
+
+@router.get("/")
+async def view_notifications(request: Request, db: Session = Depends(get_db)):
+    user_id = 1
+    return templates.TemplateResponse("notification.html", {
+        "request": request,
+        "notifications": get_unread_notifications(session=db, user_id=user_id),
+    })
+
+
+@router.post("/accept_invitations")
+async def accept_invitations(
+        request: Request,
+        db: Session = Depends(get_db)
+):
+    data = await request.form()
+    invite_id = list(data.values())[0]
```

Comment:
Is it possible the list will be empty?

Reply:
Nope.
except if someone sends an external post request, do you think I need to handle this scenario?

Reply:
at least try/except and log.error :)